### PR TITLE
Refactor the Memo field into a legacy memo and ng memo

### PR DESCRIPTION
**Pull request by hansl**
This follows the many-rs PR https://github.com/liftedinit/many-rs/pull/201 and the specification latest master and its follow up PR https://github.com/many-protocol/specification/pull/58 which includes burnt fields (to add new fields instead of re-typing old ones).

### CODE REVIEW

File: src/many-ledger/tests/migration_/memo.rs
Comment (fmorency) [inline src/many-ledger/tests/migration_/memo.rs:1]: nit: I don't like the `migration_` folder name

**Review by fmorency — CHANGES_REQUESTED** [review]
I can't run coverage because tests are failing.

File: src/many-ledger/src/migration/memo.rs
Comment (hansl) [reply src/many-ledger/src/migration/memo.rs:92]: I... uh... thought there would be more to do lol.

File: src/many-ledger/src/migration/memo.rs
Comment (hansl) [reply src/many-ledger/src/migration/memo.rs:74]: The iterator is sorted by keys. Added a comment

**Comment by fmorency** [comment]
I think we need to migrate `InfoReturn` as well. I marked the fields requiring your attention with **. The Memo is still located in the old field after the migration for legacy transactions, which is not consistent with the rest.

Everything is fine for transactions submitted **after** the migration ran.

**Before migration**
```
InfoReturn {
**  memo_: Some(
**      Memo(
**          "Legacy_Memo",
**      ),
**  ),
    transaction: Send(
       ...
    data_: None,
    state: Pending,
**  memo: None,
}
```
```
{
    0: 3,
...
            2: {
                0: [9, [1, 0]],
                1: 10000_1(h'01054d5b7773d51b36c65324baddfd3b5bc8dbb8929270b8708e479502'),
                2: 10000_1(h'80e8acda9634f6f745be872b0e5e9b65b1d3624a3ba91c6432143f60e9000000'),
**              3: "Legacy_Memo",
           ...
                9: null,
**              10: null,
            },
        },
    ],
}
```

**After migration**
```
InfoReturn {
**  memo_: Some(
**      Memo(
**          "Legacy_Memo",
**      ),
**  ),
    transaction: Send(
      ...
    data_: None,
    state: Pending,
**  memo: None,
}
```
```
{
    0: 3,
   ...
            2: {
                0: [9, [1, 0]],
                1: 10000_1(h'01054d5b7773d51b36c65324baddfd3b5bc8dbb8929270b8708e479502'),
                2: 10000_1(h'80e8acda9634f6f745be872b0e5e9b65b1d3624a3ba91c6432143f60e9000000'),
**              3: null,
                4: {
                    0: [6, 0],
       ...
                7: 1(1669302311_2),
                8: false,
                9: null,
**              10: ["Legacy_Memo"],
            },
        },
    ],
}
```

File: src/many-ledger/src/storage/event.rs
Comment (fmorency) [inline src/many-ledger/src/storage/event.rs:142]: I recall asking this question before but I don't get what this is supposed to do.

Why not just `options.set_iterate_range(MULTISIG_TRANSACTIONS_ROOT.to_vec()..)` or something?

File: src/many-ledger/tests/common/mod.rs
Comment (fmorency) [inline src/many-ledger/tests/common/mod.rs:141]: Don't we disallow using `eprintln!`?

**Review by fmorency — APPROVED** [review]
LGTM assuming comments are addressed.

I would prefer to see the new migration tests in the resiliency test instead of in the memo migration test. WDYT?

File: src/many-ledger/src/storage/event.rs
Comment (hansl) [reply src/many-ledger/src/storage/event.rs:142]: Done with something better even. Thanks for pointing me towards this, haven't seen this method.

File: src/many-ledger/tests/common/mod.rs
Comment (hansl) [reply src/many-ledger/tests/common/mod.rs:141]: The problem is we cannot pass in a trace subscriber. Also, clippy didn't complain :/

I'll add a allow above it with a comment.

**Comment by hansl** [comment]
Waiting for CI to be green, then merging https://github.com/liftedinit/many-rs/pull/201 then updating this then merging this.

**Pull request merged**
